sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1168,11 +1258,26 @@ private Task RenewOwnershipAsync()
                     ownership.ETag
                 ));
 
-            // We cannot rely on the ownership returned by ClaimOwnershipAsync to update our InstanceOwnership dictionary.
-            // If the user issues a checkpoint update, the associated ownership will have its eTag updated as well, so we
-            // will fail in claiming it here, but this instance still owns it.
+            try
+            {
+                // Dispose of all previous partition ownership instances and get a whole new dictionary.
 
-            return StorageManager.Value.ClaimOwnershipAsync(ownershipToRenew);
+                InstanceOwnership = (await StorageManager.ClaimOwnershipAsync(ownershipToRenew, cancellationToken)
+                    .ConfigureAwait(false))
+                    .ToDictionary(ownership => ownership.PartitionId);
+            }
+            catch (Exception ex)
+            {
+                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+                // If ownership renewal fails just give up and try again in the next cycle.  The processor may
+                // end up losing some of its ownership.
+
+                var errorEventArgs = new ProcessErrorEventArgs(null, Resources.RenewOwnershipOperation, ex, cancellationToken);
```

Comment:
(extreme) nit:  It may be worth going with a pattern of `Operation-XXX` or similar for these in the resources, so that they get clumped together in the alphabetical listing for intellisense and search.    That said, it's definitely not something to hold this PR up for.